lib/elixir/lib/task/supervised.ex

```diff
@@ -200,44 +202,51 @@ defmodule Task.Supervised do
        when max == 0
        when next == :done do
     %{monitor_pid: monitor_pid, monitor_ref: monitor_ref,
-      timeout: timeout, on_timeout: on_timeout} = config
+      timeout: timeout, on_timeout: on_timeout, ordered: ordered?} = config
 
     receive do
       # The task at position "position" replied with "value". We put the
       # response in the "waiting" map and do nothing, since we'll only act on
       # this response when the replying task dies (we'll notice in the :down
       # message).
-      {{^monitor_ref, position}, value} ->
-        %{^position => {pid, :running}} = waiting
-        waiting = Map.put(waiting, position, {pid, {:ok, value}})
+      {{^monitor_ref, position}, reply} ->
+        %{^position => {pid, :running, value}} = waiting
+        waiting = Map.put(waiting, position, {pid, {:ok, reply}, value})
         stream_reduce({:cont, acc}, max, spawned, delivered, waiting, next, config)
 
       # The task at position "position" died for some reason. We check if it
       # replied already (then the death is peaceful) or if it's still running
       # (then the reply from this task will be {:exit, reason}). This message is
       # sent to us by the monitor process, not by the dying task directly.
       {kind, {^monitor_ref, position}, reason} when kind in [:down , :timed_out] ->
-        waiting =
+        result =
           case waiting do
             # If the task replied, we don't care whether it went down for timeout
             # or for normal reasons.
-            %{^position => {_, {:ok, _} = ok}} ->
-              Map.put(waiting, position, {nil, ok})
+            %{^position => {_, {:ok, _} = ok, _}} ->
+              ok
             # If the task exited by itself before replying, we emit {:exit, reason}.
-            %{^position => {_, :running}} when kind == :down ->
-              Map.put(waiting, position, {nil, {:exit, reason}})
+            %{^position => {_, :running, _}} when kind == :down ->
+              {:exit, reason}
             # If the task timed out before replying, we either exit (on_timeout: :exit)
             # or emit {:exit, :timeout} (on_timeout: :kill_task) (note the task is already
             # dead at this point).
-            %{^position => {_, :running}} when kind == :timed_out ->
+            %{^position => {_, :running, value}} when kind == :timed_out ->
               if on_timeout == :exit do
                 stream_cleanup_inbox(monitor_pid, monitor_ref)
                 exit({:timeout, {__MODULE__, :stream, [timeout]}})
```

Comment:
Now that we have the value, let's do: `exit({:timeout, {__MODULE__, :stream, [value, timeout]}})`